modules/dnn/test/test_layers_1d.cpp

```diff
@@ -12,45 +12,913 @@
+    input = Mat(dims, input_shape.data(), CV_32F, inp_value);
+
+    float out_value = std::log(1 + std::exp(inp_value));
+    cv::Mat output_ref(dims, output_shape.data(), CV_32F, out_value);
+
+    std::vector<Mat> inputs{input};
+    std::vector<Mat> outputs;
+
+    runLayer(layer, inputs, outputs);
+    ASSERT_EQ(shape(output_ref), shape(outputs[0]));
+    normAssert(output_ref, outputs[0]);
+}
+
+TEST_P(Layer_Test, SoftSign)
+{
+    LayerParams lp;
+
+    lp.type = "Softsign";
+    lp.name = "softsignLayer";
+    Ptr<Layer> layer = SoftsignLayer::create(lp);
+
+    float inp_value = 1 + static_cast <float> (rand()) / (static_cast <float> (RAND_MAX/(10-1)));
+    input = Mat(dims, input_shape.data(), CV_32F, inp_value);
+
+    float out_value = inp_value / (1 + std::abs(inp_value));
+    cv::Mat output_ref(dims, output_shape.data(), CV_32F, out_value);
+
+    std::vector<Mat> inputs{input};
+    std::vector<Mat> outputs;
+
+    runLayer(layer, inputs, outputs);
+    ASSERT_EQ(shape(output_ref), shape(outputs[0]));
+    normAssert(output_ref, outputs[0]);
+}
+
+TEST_P(Layer_Test, CELU)
+{
+    LayerParams lp;
+
+    lp.type = "CELU";
+    lp.name = "celuLayer";
+    lp.set("alpha", 1.0);
+    Ptr<Layer> layer = CeluLayer::create(lp);
+
+    float inp_value = 1 + static_cast <float> (rand()) / (static_cast <float> (RAND_MAX/(10-1)));
+    input = Mat(dims, input_shape.data(), CV_32F, inp_value);
+
+    float out_value = inp_value < 0 ? std::exp(inp_value) - 1 : inp_value;
+    cv::Mat output_ref(dims, output_shape.data(), CV_32F, out_value);
+
+    std::vector<Mat> inputs{input};
+    std::vector<Mat> outputs;
+
+    runLayer(layer, inputs, outputs);
+    ASSERT_EQ(shape(output_ref), shape(outputs[0]));
+    normAssert(output_ref, outputs[0]);
+}
+
+TEST_P(Layer_Test, HardSigmoid)
+{
+    LayerParams lp;
+
+    lp.type = "HardSigmoid";
+    lp.name = "hardSigmoidLayer";
+    Ptr<Layer> layer = HardSigmoidLayer::create(lp);
+
+    float inp_value = 1 + static_cast <float> (rand()) / (static_cast <float> (RAND_MAX/(10-1)));
+    input = Mat(dims, input_shape.data(), CV_32F, inp_value);
+
+    float out_value = std::max(0.0f, std::min(1.0f, 0.2f * inp_value + 0.5f));
+    cv::Mat output_ref(dims, output_shape.data(), CV_32F, out_value);
+
+    std::vector<Mat> inputs{input};
+    std::vector<Mat> outputs;
+
+    runLayer(layer, inputs, outputs);
+    ASSERT_EQ(shape(output_ref), shape(outputs[0]));
+    normAssert(output_ref, outputs[0]);
+}
+
+TEST_P(Layer_Test, SELU)
+{
+    LayerParams lp;
+
+    lp.type = "SELU";
+    lp.name = "seluLayer";
+    lp.set("alpha", 1.6732631921768188);
+    lp.set("gamma", 1.0507009873554805);
+    Ptr<Layer> layer = SeluLayer::create(lp);
+
+
+    input = Mat(dims, input_shape.data(), CV_32F, inp_value);
+
+    float value = 1.0507009873554805 * (inp_value > 0 ? inp_value : 1.6732631921768188 * (std::exp(inp_value / 1.0) - 1));
```

Comment:
Value is out of float precision

Reply:
what to do with is?

Reply:
Performing operation on double and cast to float before compare.